spec/models/user_spec.rb

```diff
@@ -33,14 +32,19 @@
     context '.after_save' do
       let!(:user) { create(:user) }
 
-      before do
-        user.accounts.create({ name: 'Emprego CLT', active: false }, CltWorkerAccount)
-        user.accounts = [user.accounts.not.active.first]
-        user.save
+      context 'with accounts' do
+        before { user.accounts.create(CltWorkerAccount.default_build_hash) }
+
+        it { expect(user.accounts.size).to eq 2 }
+        it { expect(user.current_account).not_to be_nil }
       end
 
-      it { expect(user.accounts.size).to eq 1 }
-      it { expect(user.current_account).not_to be_nil }
+      context 'without accounts' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.